src/Build/Instance/ProjectInstance.cs

```diff
@@ -2031,6 +2020,27 @@ void ITranslatable.Translate(ITranslator translator)
             }
         }
 
+        private void MaybeForceTranslateEntireStateMode()
```

Comment:
This is breaking change. Before, everybody who read the TranslateEntireState property (it's public, so anyone can read it) got the Traits switch logic. Now, it's only happening here in translate. If CPS is using the property to derive other logic from it, their stuff will stop working correctly if they set the Trait to enforce a certain value.

Reply:
Is there a use case why would CPS like to play with this particular Trait to enforce it? To me it looks rather risky thing to do. I was under impression that this Trait is there only for internal testing and verifications.

Reply:
Having espace hatch enforcing rules as part of that getter is problematic as It effects both serialization and deserialization. I believe that expected format in deserialization shall be dictate by deserialization stream content only. For example If one node starts with escape hatch and other do not, their communication contract will be broken resulting in wide range of unexpected exceptions such as OutOfMemoryException or unexpected system behavior.

Reply:
This is within a build, right? I imagine someone would either want the escape hatch or not.

Reply:
>If one node starts with escape hatch and other do not

Currently escape hatches are activated via environment variables which are inherited by child processes. Since out of proc nodes are spawned by the build manager process, they will always share the same escape hatches right?

Reply:
@rokonec pointed out that you could build once with one set of environment variables, then change them and build again but connect to processes left over from the first build.

Reply:
Aha. Yeah, that would be a problem. I think the initial design of the out of proc nodes didn't envision for carrying over so much state. As the nodes do start to carry over more and more state, maybe we should define a more first class / sturdy cache invalidation infrastructure. For example, if we want correctness w.r.t. stuff in Traits, we should make it hashable and include that hash in the handshake.

Reply:
I have scanned github and devdiv AzDO for usage of 'TranslateEntireState'. Apart from MSBuild usage and refs found only two occurences of projects system setting this property to true.
With public interface there is always non zero chance we will break someone, but I believe that the risk in this particular case is acceptable.
I recommend to keep code as is in this PR. I recommend to keep that escape hatch as it might have tactical value for troubleshooting or experimenting.